### ISSUE TITLE
File ULIMITs for Linux containers

### ISSUE THREAD

**Issue by dwrightpulse** [linuxkit/lcow#34]
Hi,

I'm working on a PoC build to run Linux and Windows-based containers side by side on WS2019.  To enable the side by side nature I have deployed the WS2019 host as per the instructions at https://github.com/linuxkit/lcow, using dockerd.exe and docker.exe from the master project from 13/02/19 and the v4.14.35-v0.3.9 release for the initrd.img and kernel files.

I can successfully run Linux containers, but when running the influxdb:1.7.4 container from https://hub.docker.com/_/influxdb, and writing to the database the console output report that it is unable to write data due to too many open files in use.  We use this image in Azure ACI and have no issues with it. 

Having done some investigation, the ULIMIT for files in the running container when running in LCOW on the 2019 host are 1024 and when running in ACI or Docker Desktop on Win 10-1809, the limits are over 1 million (see attached screenshots)

![ulimit influx in azure](https://user-images.githubusercontent.com/44645666/53626657-42cd6600-3bfe-11e9-937c-9304d8f1e03e.PNG)
![influx ulimit in docker desktop](https://user-images.githubusercontent.com/44645666/53626676-4cef6480-3bfe-11e9-8990-1bdbfd5781d7.PNG)
![ulimit influx in docker lcow on ws2019](https://user-images.githubusercontent.com/44645666/53626786-90e26980-3bfe-11e9-9344-1f4b12997d06.PNG)

I'm posting here as concluding this is an issue with the LinuxKit kernel that this solution uses.  I have tried to see if I can use alternate kernel files to confirm if this is the issue (https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/linux-containers suggested Kata and Clear Linux kernels (but I have got nowhere with these 2 options so far)) and I am unable to over-ride the default files ulimit in the daemon.json file (as per https://docs.docker.com/engine/reference/commandline/dockerd/) as when starting the dockerd daemon it reports that this is not configurable in Windows (see further attached screenshot)

![default-ulimit in docker lcow on ws2019](https://user-images.githubusercontent.com/44645666/53627932-a60cc780-3c01-11e9-9455-0c31a85f6901.PNG)

As a way of progressing, I'm happy to try the Kata or Clear kernel option if anyone can give me some guidance on how to get hold of the right files and configure LCOW to use them....

**Comment by justincormack** [comment]
That limit will need to be set in https://github.com/Microsoft/opengcs/blob/master/init/init.c so you may want to open an issue there.

**Comment by dwrightpulse** [comment]
Thanks Justin, I'll do that...

**Comment by jstarks** [comment]
We can increase the default there, but we should also fix Docker to support —ulimit and —default-ulimit when LCOW is supported. @jhowardmsft

**Comment by dwrightpulse** [comment]
@jhowardmsft @jstarks - any update on this please?

**Comment by dwrightpulse** [comment]
@jhowardmsft @jstarks - Any update on this please?

**Comment by dwrightpulse** [comment]
@jhowardmsft @jstarks - Any update on this please?

**Comment by lowenna** [comment]
Working on higher priority stuff right now. It’s on the list

**Comment by dwrightpulse** [comment]
Thanks John, any idea on timescale.  My PoC timescale is end of May, is that feasible?

**Comment by lowenna** [comment]
I can’t provide any timescale.

**Comment by dwrightpulse** [comment]
OK, can you give any input on using alternate kernels within LCOW to see if this circumvents the issue?

**Comment by lowenna** [comment]
Not really. I was going to look at fixing it from moby not via an alt kernel

**Comment by dwrightpulse** [comment]
@jhowardmsft - are you any closer to fixing this in Moby?

Will the Build announcements ref WSL2 have any effect on how this is implemented in the future?
